Multi-source support, new path structure

nyurik opened the pull request:
Partial implementation of the #430

* New endpoint structure:
  * `GET /` -- a placeholder for the future home page
  * `GET /catalog` -- get a list of available sources, as a list of json blobs.
    * `[{id, name, description, attribution, vector_layer}, ...]`  (some fields might be missing)
  * `GET /<id>` -- get tilejson for the given source, or a combination of sources. No `.json` extension
  * `GET /<id>/<z>/<x>/<y>` -- get a tile. No format extension.
  * `GET /health` -- healthcheck
* Introduce a new tile format support crate (using code from the maplibre/mbtileserve project)
* Removed the `/rpc/...` routes - all source IDs are accessed in the same way
* Can print auto-generated configuration or save it to a file
* Refactored to support multiple sources from multiple backends, with a proper naming conflict resolution

TODO:
* benchmarks need to be rewritten - they were relying on some internal structures that are no longer there. This might be done as a separate PR due to a very different internal architecture - might need to rethink benchmarking approaches.

stepankuzmin commented on README.md:
```diff
@@ -86,22 +85,19 @@ Martin provides [TileJSON](https://github.com/mapbox/tilejson-spec) endpoint for
-| `GET`  | `/healthz`                                                                       | Martin server health check: returns `200 OK`            |
+| Method | URL                                                    | Description                                             |
+|--------|--------------------------------------------------------|---------------------------------------------------------|
+| `GET`  | `/`                                                    | [List of all sources](#source-list)                     |
```
Let's use `/` for the future user interface and `/catalog` for the list of all sources. We might want to add a list of reserved keywords for that, the same way as with `/healthz`

nyurik replied:
thanks, fixed with reserved keywords, and added a `/catalog`

nyurik committed:
Reimplement multi-source support

* New endpoint structure:
  * `GET /catalog` -- get a list of available sources, as a json blob,
    with `id -> { name, description, attribution }` fields (if
available)
  * `GET /<id>` -- get tilejson for the given source, or a combination
    of sources. No `.json` extension
  * `GET /<id>/<z>/<x>/<y>` -- get a tile. No format extension.
  * `GET /health` -- healthcheck
* Introduce a new tile format support crate (using code from the
  maplibre/mbtileserve project)
* Removed the `/rpc/...` routes - all source IDs are accessed in the
  same way
* Can print auto-generated configuration or save it to a file
* Refactored to support multiple sources from multiple backends, with a
  proper naming conflict resolution

TODO:
* benchmarks need to be rewritten - they were relying on some internal
  structures that are no longer there. This might be done as a separate
  PR due to a very different internal architecture - might need to rethink
  benchmarking approaches.

stepankuzmin reviewed (approved):
Good job! 👍
